src/Collection/CollectionTrait.php

```diff
@@ -502,4 +507,19 @@ public function unfold(callable $transformer = null)
             )
         );
     }
+
+    /**
+     * Returns the closest nested iterator that can be safely traversed without
+     * losing any possible transformations.
+     *
+     * @return \Iterator
+     */
+    protected function _unwrap()
+    {
+        $iterator = $this;
+        while (get_class($iterator) === 'Cake\Collection\Collection') {
```

Comment:
No, extensions to the class are usually because they will alter results in some way. This is just checking for a simple wrapper iterator with no added logic